sdks/python/apache_beam/runners/interactive/interactive_runner.py

```diff
@@ -125,6 +125,30 @@ def apply(self, transform, pvalueish, options):
 
   def run_pipeline(self, pipeline, options):
     pipeline_instrument = inst.pin(pipeline, options)
+    # The user_pipeline analyzed might be None if the pipeline given has nothing
+    # to be cached and tracing back to the user defined pipeline is impossible.
+    # When it's None, there is no need to cache including the background
+    # caching job and no result to track since no background caching job is
+    # started at all.
+    user_pipeline = pipeline_instrument.user_pipeline
+
+    background_caching_job_result = ie.current_env().pipeline_result(
+        user_pipeline, is_main_job=False)
+    if (not background_caching_job_result or
```

Comment:
Discussed offline. This block of code is a bit hard to read. Ning will do a minor refactor to make this clearer.

Reply:
Refactored this logic into a separate module that can be invoked here and in outer scopes such as `show`.